Infrastructure/Repositories/Users/RolesRepository.cs

```diff
@@ -0,0 +1,169 @@
+﻿using Infrastructure.Entities.Users;
+using Infrastructure.Repositories.Users.Interfaces;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repositories.Users
+{
+
+    public class RolesRepository : IRolesRepository
+    {
+        private readonly LANHQDbContext _context;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly RoleManager<ApplicationRole> _roleManager;
+
+        public RolesRepository(LANHQDbContext context, UserManager<ApplicationUser> userManager, RoleManager<ApplicationRole> roleManager)
+        {
+            this._context = context;
+            _userManager = userManager;
+            _roleManager = roleManager;
+        }
+
+        public async Task<IEnumerable<ApplicationRole>> GetRoles()
+        {
+            return await _context.Roles.ToListAsync();
+        }

```

Comment:
```suggestion
        public Task<IEnumerable<ApplicationRole>> GetRoles() => _context.Roles.ToListAsync();
```
L'appelant vas faire le await, nous on vas juste retourner la Task. Juste vérifier si on a pas un problème avec l'instance du context lorsqu'on fait ça.